File: go/internal/collector/service_status.go
package collector

// serviceStatusFromCode maps launchctl exit/status codes to Java ServiceStatus names.
func serviceStatusFromCode(code int) string {
	switch {
	case code == 0:
		return "RUNNING"
	case code > 0:
		return "ERROR"
	default:
		return "STOPPED"
	}
}

File: src/main/java/org/dikkulah/instancemetriccollector/model/ServiceStatus.java
package org.dikkulah.instancemetriccollector.model;

public enum ServiceStatus {
    RUNNING,
    ERROR,
    STOPPED;

    public static ServiceStatus fromStatusCode(int statusCode) {
        if (statusCode == 0) {
            return RUNNING;
        }

        if (statusCode > 0) {
            return ERROR;
        }

        return STOPPED;
    }
}